bed/modify.go

```diff
@@ -70,3 +71,176 @@ func MergeHighMem(records []Bed, mergeAdjacent bool) []Bed {
+	SortByCoord(records)
+	var curr = records[0]
+	for i := 1; i < len(records); i++ {
+		if Overlap(curr, records[i]) {
+			if records[i].Score < curr.Score {
+				curr = records[i]
+			}
+		} else if curr.Chrom == records[i].Chrom {
+			dist = records[i].ChromStart - curr.ChromEnd
+			if (curr.Score + dist) < records[i].Score { //if the distance to the record on the left plus its hidden score
+				// is less than the hidden score stored for the right record, the right record will be reassigned the
+				//hidden that is equal to the dist to the left record plus it's hidden value and the gene name stored on the left record
+				newHidden = true
+				records[i].Score = curr.Score + dist
+				records[i].Name = curr.Name
+			} else if (records[i].Score + dist) < curr.Score {
+				curr.Score = records[i].Score + dist
+				curr.Name = records[i].Name
+			}
+			outList = append(outList, curr)
+			curr = records[i]
+		} else {
+			outList = append(outList, curr)
+			curr = records[i]
+		}
+	}
+	outList = append(outList, curr)
+	return outList, newHidden
+}
+
+func removeRecordsOnEmptyChrom(records []Bed, genome map[string]chromInfo.ChromInfo) []Bed {
+	var out = make([]Bed, 0)
+	var exist bool
+
+	for i := range records {
+		_, exist = genome[records[i].Chrom]
+		if exist {
+			out = append(out, records[i])
+		}
+	}
+	return out
+}
+
+// FillSpaceNoHiddenValue accepts a bed and reference genome as a map[string]chromInfo.ChromInfo and returns a bed
+// that assigns each genomic position to the nearest feature in the input bed using the Name field in the input bed.
+// Absolute start positions of the original input bed entries are stored in the Score field of the output bed entries.
+func FillSpaceNoHiddenValue(records []Bed, genome map[string]chromInfo.ChromInfo) []Bed {
+	records, _ = mergeKeepLowNameAndScore(records)
+	records = removeRecordsOnEmptyChrom(records, genome)
+	var answer = make([]Bed, 0)
+	if records == nil || len(records) == 0 {
+		return records // return slice as is if it is empty or nil.
+	}
+
+	var currAnswer = Bed{Chrom: records[0].Chrom, ChromStart: 0, ChromEnd: records[0].ChromEnd, Name: records[0].Name, Score: records[0].ChromStart, FieldsInitialized: 5}
+	var midpoint int
+
+	//when a position is equidistant from two features, it is assigned to the left feature.
+	for i := 1; i < len(records); i++ {
+		if records[i].Chrom != currAnswer.Chrom {
+			currAnswer.ChromEnd = genome[currAnswer.Chrom].Size
+			answer = append(answer, currAnswer)
+			currAnswer.Chrom = records[i].Chrom
+			currAnswer.ChromStart = 0
+			currAnswer.ChromEnd = records[i].ChromEnd
+			currAnswer.Name = records[i].Name
+			currAnswer.Score = records[i].ChromStart
+		} else {
+			midpoint = (records[i].ChromStart + currAnswer.ChromEnd) / 2
+			currAnswer.ChromEnd = midpoint + 1
+			answer = append(answer, currAnswer)
+			currAnswer.Chrom = records[i].Chrom
+			currAnswer.ChromStart = midpoint + 1
+			currAnswer.ChromEnd = records[i].ChromEnd
```

Comment:
Based on the last few lines, I think things might get funny here (we end up with invalid beds) if more than one TSS comes in with the same location (which will probably happen).  There is a test for that or is that something we know we handle?

Reply:
If I remember correctly, the input to this function should already have been filtered such that no more than one TSS has the same location?

Reply:
This is fine but will need documentation